### ISSUE TITLE
v2.2.0 New UI

### ISSUE THREAD

**Issue by cleidigh** [cleidigh/printing-tools-ng#159]

**Comment by kiki-ja** [comment]
@cleidigh
To All

PTNG 2.2.0-b9
Thunderbird 102.7.0 64bit
Windows 10 Pro 22H2 64bit

#157 mentions.

I have created a UI test version translated into Japanese.
- [ptng-options.xhtml] I translated only the untranslated parts into Japanese.
- [ptng-options.css] I changed the color of 'PDFoutputDir' to no color.

Result:.
In the beta version, we found that when localizing to Japanese, it is necessary to specify the width size of the translated part of TAB5.
The reason for this is that the layout is broken.
It would be helpful if the release version does not require width size specification for any language.

Please see screenshots.

![220b9-kiki-tab1](https://user-images.githubusercontent.com/80428201/213925517-1e9d3751-5b02-4ccf-a71a-e1ab6fd4edab.png)

![220b9-kiki-tab2](https://user-images.githubusercontent.com/80428201/213925523-5e3516e8-d086-4885-9814-cc6d782ef1df.png)

![220b9-kiki-tab3](https://user-images.githubusercontent.com/80428201/213925529-e0c00952-efa9-4846-8bf3-37b990452a65.png)

![220b9-kiki-tab4](https://user-images.githubusercontent.com/80428201/213925539-503fb20a-5f14-49af-9286-bcb4e50c46da.png)

![220b9-kiki-tab5](https://user-images.githubusercontent.com/80428201/213925546-b3b2bc01-5c39-403b-bd20-cf7a6a765866.png)

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
posted  b11
ui translations and still working on table for tab5.
See if you can verify what css widths are good.
Don't do other alignments until we get the widths set.
The table is very tricky...
@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

I have installed PTNG 2.2.0-b11.

There are corrections to the translated parts of the UI.
I will inform you separately.

CSS = blue border section.
- A little wide. I feel that line breaks would be a good. TAB1, 4 
- A little narrow. TAB5

About the red border.
- Header and Footer 'Center(中央)' is vertical. TAB5

Please see screenshots.

**[TAB1]**
![220b11-ja-tab1](https://user-images.githubusercontent.com/80428201/213974419-aff9cf4a-56d3-477a-b879-b4a68b38f7a1.png)

**[TAB4]**
![220b11-ja-tab4](https://user-images.githubusercontent.com/80428201/213974682-b4ac43de-76f3-4bfa-911a-63c32332f0f7.png)

**[TAB5]**
![220b11-ja-tab5](https://user-images.githubusercontent.com/80428201/213974458-7aa95740-d033-4182-aaf4-1048f4e1b4bc.png)

>The table is very tricky...

I too find html table tricky (TAB5).
It might be better to specify the width in ptng-options.xhtml.

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
I stopped development earlier today so I had time to write you some more useful notes.

Translations 
- I reviewed your changes and I used your dtd with the addition of page_headers
- Fixed some de translations 
- I will make a list of the last set of translations 

UI Input widths
- I reset all the widths to be equal in all locales
- They look pretty good to me now that we have everything translated 
- If you think an input length should change then change the ja width in ptng-options.css
- If we determine the lengths can be equivalent, then we will remove rules

Table for page options 
- I am very pleased as I got everything really nicely aligned and spaced well 
- The table allows everything to be responsive while using columns to keep alignment
- Since many of the labels in Japanese are one or two characters, we can change the padding which stretches distances between cells
- I will show the table with borders so you can understand the layout better 

I am happy with the en layout and I think the ja layout is also good assuming you like the table spacing. I think we need to finalize the ui layout asap so we can get this out. I would love to release by end of January if possible.

<img width="407" alt="2023-01-23-17-54-18 " src="https://user-images.githubusercontent.com/25272315/214172057-a7355954-5f44-4a0e-a388-21593715f3aa.png">

<img width="407" alt="2023-01-23-17-54-43 " src="https://user-images.githubusercontent.com/25272315/214172362-c8d70768-8047-4d31-a8ab-cbdab7ef4c4e.png">

<img width="441" alt="2023-01-23-17-57-09 " src="https://user-images.githubusercontent.com/25272315/214172457-099d8aa3-6db9-4a48-9d6e-a4fbaec64340.png">

<img width="441" alt="2023-01-23-17-57-27 " src="https://user-images.githubusercontent.com/25272315/214172527-6131ba37-7c95-43d1-94fe-aa547a2b7f3a.png">

<img width="441" alt="2023-01-23-17-57-53 " src="https://user-images.githubusercontent.com/25272315/214172638-2922c95a-af4c-4d8b-ae9d-9a0c331fb818.png">

<img width="441" alt="2023-01-23-18-02-28 " src="https://user-images.githubusercontent.com/25272315/214172723-48f8a846-be88-46d3-ae3a-5ba4c23447a6.png">

Pretty cool huh?
@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

>Translations
>I reviewed your changes and I used your dtd with the addition of page_headers

I have checked.
Thanks for the corrections and additions.

>UI Input widths
>I reset all the widths to be equal in all locales

Unfortunately, I did not get the same widths in my case.

>If you think an input length should change then change the ja width in ptng-options.css

Could you expand the input lenght of TAB1, 4, and 5 a bit?
ptng-options.css:
TAB1:#customDate:lang(ja) 280 --> 390
TAB4:#customDatePDF:lang(ja) 280 --> 400
TAB4:#PDFcustomFilenameFormat:lang(ja) 364 --> 390
TAB5:I do not know table width.

The reason is that the string you entered is missing and unreadable.
See the blue border in the screenshot.

[TAB1]
![220b13-ja-tab1-0](https://user-images.githubusercontent.com/80428201/214204933-dfc7bd43-f0a9-492e-bf6f-7131adf7fba1.png)
[TAB4]
![220b13-ja-tab4-0](https://user-images.githubusercontent.com/80428201/214204944-7664fad0-2bb0-4819-b0c9-12ab2d943f5a.png)
[TAB5]
![220b13-ja-tab5-0](https://user-images.githubusercontent.com/80428201/214204955-3c2cbbfd-90ca-4d37-8a0d-11af395aebee.png)

>Pretty cool huh?

Yes, it has improved a lot.
I had created the same table.

I will add a separate comment on the UI detail request.

@kiki-ja

**Comment by RRoenn** [comment]
you do a lot on ui improvement
 may I suggest these changes for alignment?

![grafik](https://user-images.githubusercontent.com/77779094/214254447-d39ad758-2048-49e2-8b33-3f898497279c.png)

I prefer the compact version but if you change the table this would be my suggestion

![grafik](https://user-images.githubusercontent.com/77779094/214255481-732049a3-7d39-454e-80d3-2fdee6ddb840.png)


Is it correct to translate Okay/Cancel to Ok in Japanese? No Japanese Characters?

**Comment by kiki-ja** [comment]
@RRoenn 

The current 2.2.0-b13 version has different window sizes and input lenght for the English and Japanese versions.
This is due to the difference between 1-byte English characters and 2-byte Japanese characters, and the difference in OS system font size.
e.g 
a=1byte
あ=2byte

Therefore, Japanese requires a slightly wider input lenght than English.
Please see screenshots.
**[TAB4]**
![TAB4 en-US ja diff](https://user-images.githubusercontent.com/80428201/214275557-90da54a2-3478-4cfd-9fe4-87966862bb10.png)
**[TAB5]**
![TAB5 en-US ja diff](https://user-images.githubusercontent.com/80428201/214275583-28ad559f-a71f-4d64-891c-8d719dc32a24.png)

@RRoenn you are proposing an English version?

>Is it correct to translate Okay/Cancel to OK in Japanese? No Japanese Characters?

Yes, OK is commonly used in Japanese, and so is the Thunderbird option.
It is more difficult to translate the meaning into Japanese any longer.
Please see screenshots.

![OK - Cancel](https://user-images.githubusercontent.com/80428201/214275759-f820148b-dc91-4f80-bf22-4ed4ec4dde99.png)

@kiki-ja

**Comment by RRoenn** [comment]
> @RRoenn you are proposing an English version?

no, sorry for confusing, following compact/short version, any kind of language ...
![grafik](https://user-images.githubusercontent.com/77779094/214294413-a8658c48-8da6-4dc9-8aa9-4f6250fa5ef0.png)

and longer/wider version
![grafik](https://user-images.githubusercontent.com/77779094/214255481-732049a3-7d39-454e-80d3-2fdee6ddb840.png)

**Comment by cleidigh** [comment]
@kiki-ja 
@RRoenn 
I am constrained by some limitations having lots of legacy XUL and html tables etc.
For b14 it's a balance to get multiple locales to look good with best,but not always ideal alignments.

@kiki-ja 
I changed the three widths you requested.

@RRoenn 
I adjusted the alignments on Tab4, however, I took the approach to shift items so they don't attempt to be exactly aligned. This way with different locales they have the same feel.

@kiki-ja 
@RRoenn 
After a lot of experiments I have left the table alone.
I cannot align the page ranges as it cant be defined to go to the table edge dynamically, xul and table limitations.
Same for the header footers fields. Expanding these messes up the table.
Perhaps later I could look at a flex based layout. From a usability standpoint most people will only have one or two tokens and the fields can contain more than its length.

I really have to finish this up because I need time for IETNG and the next ESR of Thunderbird.

@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

About the PDF Output (TAB5) input fields

>I changed the three widths you requested.

It turned out to be a good size.
Thank you very much.

Please see screenshots.
![220b14-ja-tab4-good](https://user-images.githubusercontent.com/80428201/214753937-5f0b574c-367f-4ca7-814f-1bc43560f552.png)

--------
'Output PDF Directory', 'Prefix', 'Suffix' and 'Filter Characters' have garbled Japanese characters.
Can these be fixed?

Please see screenshots.
![TAB4 diff](https://user-images.githubusercontent.com/80428201/214754116-457b1607-cc10-49d8-b37b-da28909ce8f3.png)

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh

The following is my request for the UI.
I would like it to be easier to read and more beautiful.

I already told you this at b9, but I will repeat it again here.
Please forgive us.

See below, along with screenshots.

１．Place a slightly wider space above the Headers Order and a slightly smaller space below.
Make the boundaries of the item group easier to understand.

b14-ptng-options.xhtml:
Line:152  spacer height="6px" --> spacer height="15px"
Line:157  spacer height="15px" --> spacer height="8px"
![TAB2 diff](https://user-images.githubusercontent.com/80428201/214755223-7ae3e8ed-24bc-453b-88ac-ae358506ff1e.png)

２．Align the vertical columns of the checkboxes. Then, add spaces to make the boundaries of the item group easier to understand.

b14-ptng-options.xhtml:
Line:250 spacer width="5px" --> spacer width="10px
Line:262 spacer width="30px"  --> spacer width="35px"
Line:275 spacer width="5px" --> spacer width="10px"
![TAB3 diff](https://user-images.githubusercontent.com/80428201/214755264-b26d3065-ec0c-451c-afd8-2d759e2a642c.png)

３．Only TAB5 has a space at the top, and its height is not aligned with other TABs.
Also, it would be better to put a little wider space at the boundary of the item group for better visibility.

b14-ptng-options.xhtml:
Line:354 spacer height="6px" --> delete
Line:361 spacer width="20px" --> spacer width="10px"
![TAB1-5 diff](https://user-images.githubusercontent.com/80428201/214755306-729e431e-e84c-4bdc-9ed4-5b7c3fc7059a.png)

４．Align the vertical columns of the printer selection dialog.
Make the header and footer input lenght (width) a bit wider.
I am not sure about <html:table>.
The only way I can think of is to insert style="width: *px".

b14-ptng-options.xhtml:
Line:361  spacer width="20px" --> spacer width="15px"
![TAB5 diff](https://user-images.githubusercontent.com/80428201/214755329-897cec6a-c096-4f99-b463-13c5e68fa8f7.png)

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 

b15 - tweaks and test - some functions broken, partial translation stuff

UI spacing tweaks: 

I made the spacing tweaks to Tabs 2,3 and 5
I changed the spacing before dropdown because it was wrong 

The Table:

The static widths in the table are really not the way to do this. Any change in elements breaks the alignments.
I did testing with just the first header row.
This approach will require css width rules for any locale.
I have test rules for en, ja and de 
I also realized that the alignments will probably be off because of a small difference between in and mm units , an example of why this is a bad way.

So it be interesting if the input is aligned.
@kiki-ja and @RRoenn can you check if the first header row aligns with the end of the margin row? All three locales are aligned for me.
@cleidigh

**Comment by cleidigh** [comment]
@kiki-ja 
Concerning the various odd fields. These should have all started with their defaults as empty strings. Can you reset them to empty strings in the preferences editor or the ui?
@cleidigh

**Comment by cleidigh** [comment]
@kiki-ja 
These are the last translations I believe...
/ { key: "", text: "" },
	{ key: "confirm_pdf_autosave", text: "Click OK to Autosave PDF or Cancel to Preview" },
	{ key: "select_pdf_dir", text: "Select PDF Output Directory" },
	{ key: "writing", text: "Writing" },
	{ key: "printing", text: "Printing" },
	{ key: "err_copies_val_req", text: "Copies value required" },
	{ key: "err_pageranges_val_req", text: "Page Ranges value required" },
	{ key: "err_pageranges_val_notzero", text: "Page Ranges value cannot be zero" },
	{ key: "err_pageranges_val_endgrbeg", text: "Page Range end must be greater than start" },

@cleidigh

**Comment by RRoenn** [comment]
> @RRoenn can you check

sorry, I am on holiday till Tuesday (without PC/Laptop)
@RRoenn

**Comment by kiki-ja** [comment]
@cleidigh

>UI spacing tweaks:
>
>I made the spacing tweaks to Tabs 2,3 and 5
>I changed the spacing before dropdown because it was wrong

Thank you very much.
I have edited many of the vertical and horizontal spacers to further fine-tune the customization.
The TAB5 header and footer have not been edited.

Attached are the ptng-options.xhtml and ptng-options.css that I edited.
What do you think?

![220b15-kiki-tab1-0](https://user-images.githubusercontent.com/80428201/215100594-d65a3daf-7d79-4eff-987a-28954d76a4bf.png)

![220b15-kiki-tab2-0](https://user-images.githubusercontent.com/80428201/215100625-c2e8e727-6d19-493b-9540-1b01c8da4fd9.png)

![220b15-kiki-tab3-0](https://user-images.githubusercontent.com/80428201/215100657-17b6bb7e-6bd6-40a6-a19c-db6c92b1ce72.png)

![220b15-kiki-tab4-0](https://user-images.githubusercontent.com/80428201/215100685-28a56012-2694-4ac4-9977-c1b08f6fdbe7.png)

![220b15-kiki-tab5-0](https://user-images.githubusercontent.com/80428201/215100750-e0cef5c8-74ee-49ea-83e4-a8b7c85a6ad4.png)

**Attachment:**
[b15-options-xhtml_css.zip](https://github.com/cleidigh/printing-tools-ng/files/10519405/b15-options-xhtml_css.zip)

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh

>The Table:
>
>The static widths in the table are really not the way to do this. Any change in elements breaks the alignments.
>I did testing with just the first header row.

The results are as follows.
This is a difficult problem.

![220b15-ja-tab5](https://user-images.githubusercontent.com/80428201/215102615-683db623-35db-409c-8712-fc7aa53c43cc.png)

>@kiki-ja and @RRoenn can you check if the first header row aligns with the end of the margin row? All three locales are aligned for me.

As shown in the screenshot above.

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh

>Can you reset them to empty strings in the preferences editor or the ui?

I have reset the following in about:config to blank
'Output PDF Directory', 'Prefix', 'Suffix' and 'Filter Characters'.

extensions.printingtoolsng.pdf.output_dir
extensions.printingtoolsng.pdf.filename.prefix
extensions.printingtoolsng.pdf.filename.suffix
extensions.printingtoolsng.pdf.filename.filter_characters

As a result, garbled Japanese characters were not improved.

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh
To All.

**About the TAB5 (Print and Page Options) table:**

We find it difficult to align the right edge of the input width and unify the width due to differences between inches and meters, system font differences, locale differences, etc.

We suggest that only the left edge of each item be aligned and the right edge be compromised with the required length of input width position.

I think the Japanese locale is fine with the compact customization I suggested/requested.
note: excludes headers and footers.

The margins are already nicely aligned left and right, so that is not a problem.
I think it would be acceptable if the headers and footers widths were not only '&*', but also if there were widths that allowed arbitrary text to be entered.

What do you think?

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
I merged in your spacing edits. I did maintain a larger distance to the filename format, I want that more separated since it is the culmination of the above options.

On the preferences issue : 

I found i had those preferences with the wrong type in one area and not the others. Fixed for the next beta tomorrow.

Translations : 
I have an updated list below.
The "Writing" and "Printing" are both used in the status line for pdf output. They will look like this:

Writing: Some msg name.pdf

Note i do not include the colon in the string.

On the table: 
I will do some more experiments tomorrow and post a couple of potential cases. I agree having some space for one or two words is good.
I think we will end up with a "ragged" right edge. I think this will look ok.  Sometimes in layouts it's not always good to align everything.

Should look good tomorrow!

// { key: "", text: "" },
	{ key: "confirm_pdf_autosave", text: "Click OK to Autosave PDF or Cancel to Preview" },
	{ key: "select_pdf_dir", text: "Select PDF Output Directory" },
	{ key: "writing", text: "Writing" },
	{ key: "printing", text: "Printing" },
	{ key: "err_copies_val_req", text: "Copies value required" },
	{ key: "err_copies_val_notzero", text: "Copies value cannot be zero" },
	{ key: "err_pageranges_val_req", text: "Page Ranges value required" },
	{ key: "err_pageranges_val_notzero", text: "Page Ranges value cannot be zero" },
	{ key: "err_pageranges_val_endgrbeg", text: "Page Range end must be greater than start" },

@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

>I merged in your spacing edits. I did maintain a larger distance to the filename format, I want that more separated since it is the culmination of the above options.

I understand.
[TAB4 Custom sample]
![220b15-kiki-tab4](https://user-images.githubusercontent.com/80428201/215265969-14a2716b-310c-4ffd-b726-87f0e036b593.png)

>On the preferences issue :
>I found i had those preferences with the wrong type in one area and not the others. Fixed for the next beta tomorrow.

Thank you very much.

>Translations :
>I have an updated list below.
>The "Writing" and "Printing" are both used in the status line for pdf output.

Thank you very much.
I am not sure because I have not actually tested it, but I will try to translate it.

>On the table:
>I think we will end up with a "ragged" right edge. I think this will look ok. Sometimes in layouts it's not always good to align everything.

I agree.
It would be ideal for right edge to be align, but not inevitable.

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
b16 has your spacing edits and the full property translations.
AND...
Last night I had an idea and it looks like I found a way to get the table alignment!

I determined that if I query the initial table cell size for the first header I can then use that to set the input width. Some more css and on my system all locales align.
This should do the same for you as it's totally responsive to all the things that can affect sizing.

Christopher

**Comment by kiki-ja** [comment]
@cleidigh

About the TAB4:
The Japanese garbled characters have been fixed and resolved.
Thank you very much.

![220b16-ja-tab4-0](https://user-images.githubusercontent.com/80428201/215472465-af5a01b2-5862-42ea-9615-9b1095df584d.png)

About the TAB5 table:
The headers and footers right edge are now nicely aligned.
Thank you very much.
I would be happy if the widths of margins A and B were the same, as shown in the screenshot below.

![220b16-ja-tab5-0](https://user-images.githubusercontent.com/80428201/215472526-93c2db75-b550-4ec2-9bb5-c116370f7569.png)

Translations:
Sorry for the delay, but I translated it.
The file is attached.

[220_ja_translation_text.zip](https://github.com/cleidigh/printing-tools-ng/files/10535591/220_ja_translation_text.zip)

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
Thank you for the translations, I have merged them in.

On the table : 

This is another issue of trying to get alignments in a responsive table.
I did my best to make A=B
on my system I can get them fairly close, I don't know how your layout will look.

check  b17
@cleidigh

**Comment by RRoenn** [comment]
TB 102.7.1 (64-Bit)
PTNG b17

Maybe there is a better German translation for body (tab3)? - I'm looking for ...
Below are the tabs for your information

![grafik](https://user-images.githubusercontent.com/77779094/216009748-8599ec3f-91f0-43a1-a6e4-d2b6cd7a2d5a.png)

![grafik](https://user-images.githubusercontent.com/77779094/216009909-83d86792-9e0c-41c2-b2da-95215a322037.png)

![grafik](https://user-images.githubusercontent.com/77779094/216009976-e43ae923-8ca5-4440-a1b7-ea6efc691edc.png)

![grafik](https://user-images.githubusercontent.com/77779094/216010183-f3277131-1715-46f9-8368-3b2e8f254ba5.png)

![grafik](https://user-images.githubusercontent.com/77779094/216010263-e8d9bd02-79e5-41b2-8986-e5f0be2a08db.png)

**Comment by kiki-ja** [comment]
@cleidigh

PTNG 2.2.0-**b17**
Thunderbird 102.7.1 ja, en-US 64bit
Windows 10 Pro 22H2 64bit

**About the TAB5 (Margins) table:**

>I did my best to make A=B
>on my system I can get them fairly close, I don't know how your layout will look.

Unfortunately, A and B are not the same size in the en-US version at my location.
The right edge is nicely aligned with the right edge of the Headers and Footers.
Please see the screenshot.

![220b17-ja-tab5-bad](https://user-images.githubusercontent.com/80428201/216092310-b56f8e34-f425-47e5-a18b-d2a99b0e9936.png)

**My compromise plan:**
My customization (ja only).
It is a little better.
ptng-options.xhtml:
Line:429 style="text-align: **right**" --> style="text-align: **center**"
Line:452 style="text-align: **right**" --> style="text-align: **center**"

However, the right edge does not align nicely with the right edge of the Headers and Footers.
Please see the screenshot.

![220b17-ja-tab5-center-0png](https://user-images.githubusercontent.com/80428201/216092842-7cb95ae8-6f40-4aea-beed-f1585c23139b.png)

**My ideal:**
I have no knowledge and skills in html:table.
So this is just a sample image.
Please see the screenshot.
Note: The numbers represent percentages.

![Margins Layout-1](https://user-images.githubusercontent.com/80428201/216093584-dda29bcb-db78-4ad8-8e77-c164e9995cb1.png)

![Headers Footers Layout-1](https://user-images.githubusercontent.com/80428201/216093647-412c8b51-2d13-41ed-acab-361e32af0e46.png)

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
Sorry I was offline yesterday unable to work.

So first on your ideal proposal. The table really cannot be defined in percentages as it removes the responsiveness and causes other issues.
I do think you can get A=B with an approach based on changing the current css.

In your first idea you changed the Alignment of the last column to center. This moved the number closer to the label column, but had the negative effect of losing the left alignment.

We can use the css rule for the forth (label) column to accomplish the reduced distance, but this does not affect left side table alignment.

So the rule starts at line # 390 in ptng-options.css : 

.margins-hdrftr-tbl td:nth-child(4) { 
	padding-left: 25px;
	padding-right: 0px;
} 

First add a right Alignment rule 
	text-align: right;
This will shift the label closer to the number. There will still be minimal space between the two. On my system this was too close and B<A.
Now you can adjust the right side padding of column 4 which will increase the distance. You should be able to make A=B
In case you are unable we may also try padding on column two, but I don't think we will need that. So your starting point should look like:


.margins-hdrftr-tbl td:nth-child(4) { 
	padding-left: 25px;
	padding-right: 0px;
	text-align: right;
}

If B<A, too close, then increase padding-right

Let me know so we can close this to finish this up.
@cleidigh

**Comment by cleidigh** [comment]
@RRoenn 
How about Nachrichtentext
I got this by translating Message body which sounds like it's correct. It will stretch things, but we can adjust if really necessary.

Repeating above...
Have you ever seen the preview give inch units when it should be mm?
@cleidigh

**Comment by RRoenn** [comment]
> @RRoenn How about Nachrichtentext I got this by translating Message body which sounds like it's correct. It will stretch things, but we can adjust if really necessary.
> 
> Repeating above... Have you ever seen the preview give inch units when it should be mm? @cleidigh

I was fine with "Header" and "Body", but that's no real german.
I was think about "E-Mail", but tab 3 is just a part from the hole E-Mail.
"Nachrichtentext" is a good translation and equal like first point on that tab 3 and also corresponding to tab 2 "Nachrichtenkopf". What do you think about **"Textteil"**?. This is shorter.

Please change "Header-Reihenfolge" in just "Reihenfolge" in tab 2 to kill the last english words.

I can't remember inch or mm, but I think always mm ...

**Comment by cleidigh** [comment]
@RRoenn 
Thanks, I like Textteil
Short and sweet!
I will make the other correction.
@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

**About TAB5 (Margins) table**

>So first on your ideal proposal. The table really cannot be defined in percentages as it removes the responsiveness and causes other issues.
>I do think you can get A=B with an approach based on changing the current css.

I understand.

**About ptng-options.css**
ptng-options.xhtml:
Reverted to html:td style="text-align: right".

[Test 1]
ptng-options.css
.margins-hdrftr-tbl td:nth-child(4) {
padding-left: **25**px;
padding-right: 0px;
**text-align: right;**
}

A little better.
Please see the screenshot.

![220b17-ja-tab5-text-align-right](https://user-images.githubusercontent.com/80428201/216742205-b7bab524-117d-4ed6-a975-c05efe5daf36.png)

[Test ２]
ptng-options.css
.margins-hdrftr-tbl td:nth-child(4) {
padding-left: **45**px;
padding-right: 0px;
**text-align: right;**
}

This is now in just the right position.
Please see the screenshot.

![220b17-ja-padding-left-45px](https://user-images.githubusercontent.com/80428201/216742218-0b1036eb-4309-45e5-a029-c965e6767bbb.png)

>On my system this was too close and B<A.

I also tried the en-US version and got the same result.
So please use only ja version for the above Test 2 CSS.

**P.S.**
Please also update printingtoolsng-help.html.

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
Excellent 
I am working on the help now
@cleidigh

**Comment by RRoenn** [comment]
Hello @cleidigh 

unfortunately, when entries are hidden, the contrast in the new version is not so good. Is this because of dark mode?

**old version**
![grafik](https://user-images.githubusercontent.com/77779094/216975593-dd40982c-c19f-48e1-8939-6e9db5497caa.png)

**new version**
![grafik](https://user-images.githubusercontent.com/77779094/216975678-a2bbbe5e-d7a5-4fd3-b7ea-64805a5bf802.png)

**Comment by RRoenn** [comment]
please change "Überschriften" to "Kopfzeilen" on tab 5

**Comment by RRoenn** [comment]
now I get (inches) in tab 5 and (mm) in preview.

![grafik](https://user-images.githubusercontent.com/77779094/216979226-9d25abac-2696-417c-805a-a462a8f567f1.png)

![grafik](https://user-images.githubusercontent.com/77779094/216979748-dd780787-f3a9-4b39-9601-5033944ba86a.png)

**Comment by RRoenn** [comment]
switching to "save as pdf" in preview

![grafik](https://user-images.githubusercontent.com/77779094/216980255-17f7a56b-7a81-45ec-98d7-82e9a65e11e0.png)

**Comment by RRoenn** [comment]
something stranges happen, if I switch printer and switch and switch ....

![grafik](https://user-images.githubusercontent.com/77779094/216981090-5a787db8-25cb-41b8-9ce1-6c1b4df6e344.png)

**Comment by RRoenn** [comment]
Following are the same printers, just with other name. First has (mm) displayed, second has (in) displayed.

![grafik](https://user-images.githubusercontent.com/77779094/216985099-65ca562d-b618-4c8d-bf38-4a90abc73c11.png)

![grafik](https://user-images.githubusercontent.com/77779094/216985504-3905dfc4-c773-4436-90a7-e694ce280911.png)

**Comment by kiki-ja** [comment]
@cleidigh
@RRoenn

PTNG 2.2.0-b17
Thunderbird 102.7.1 64bit
Windows 10 Pro 22H2 64bit

>something stranges happen, if I switch printer and switch and switch ....

[TAB 5]
I reproduced it too.
I assume the problem is with the table because it happens only in the Margins, Headers, and Footers.

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh
@RRoenn

>Following are the same printers, just with other name. First has (mm) displayed, second has (in) displayed.

The default value is different for each printer.
Check '**print_paper_size_unit**' in about:config.
0 = in
1 = mm

@kiki-ja

**Comment by cleidigh** [comment]
@RRoenn 
Can you capture the paper_size_unit for each of your printers. I believe that the drivers and Thunderbird should always use mm in your locale.
Something weird is going on. This is one of those issues I just cannot reproduce because I have never found a way to emulate another locale and region. Setting another locale here does not change the units, only the language!
I will release beta today, but its help and translations. I will make the German tweaks, thanks.
@cleidigh

**Comment by cleidigh** [comment]
@RRoenn 
@kiki-ja 
I was able to reproduce the magically expanding table. I think I already know why...
@cleidigh

**Comment by RRoenn** [comment]
> @RRoenn Can you capture the paper_size_unit for each of your printers. I believe that the drivers and Thunderbird should always use mm in your locale. Is it consistent that the second printer shows inches? What does preview show for that printer?

It's the same printer/driver, just different system settings (paper tray 1 or 2, color yes or no)
The second printer shows alway inches in PTNG settings, but the values are mm (12,7 mm are correct equals 0,5 in) and **12,7 in are WRONG**. Print preview is mm (12,7 mm).

Some strange behavior ...
second printer - print preview is correct 12,7 mm.
if I switch to build-in "save pdf" - print preview shows 0,5 in
then select second printer again - print preview shows also 0,5 in (equals 12,7 mm).
First printer always stays at 12,7 mm

crazy, i don't know why
**paper_size_unit**
![grafik](https://user-images.githubusercontent.com/77779094/217074190-6cdf0e5c-4a42-476f-ae75-eee4d9121372.png)

I had both installed on the same day

**Comment by RRoenn** [comment]
I think preview take the 12,7 mm from the PTNG settings, after switching printers in preview, it take mm/in from tb config?
Maybe the inch is just a display bug in local ui from PTNG?

I had set all 'print_paper_size_unit' to 1
![grafik](https://user-images.githubusercontent.com/77779094/217075951-939b960d-eab0-4292-afd3-10148f919d17.png)

After some printer switches in tab 5, some printer was set back to 0 ...

**Comment by RRoenn** [comment]
After some printer switches in tab 5, some printer was magical set back to 0 ...
sometimes ''print.printer_Mozilla_Save_to_PDF.print_paper_size_unit'' and sometimes
''print.printer_develop_ineo_+224_E-Mail.print_paper_size_unit'' (secund printer)
Do you change them with PTNG?

**Comment by cleidigh** [comment]
@RRoenn 
I already fixed the expanding table, that was easy...
This one is killing me as I look at the tb code I get nothing. The units should be derived from the pape, but it's not consistent.

Ok your last comment that the units change is a start. I don't change the units directly, but all settings get saved if we change anything in ptng.
Is it a problem to uninstall the inno printer clear tb prefs for that printer, reinstall and check what the units are?

I will also try to figure out some debug.
Thanks !
@cleidigh

**Comment by RRoenn** [comment]
> Is it a problem to uninstall the inno printer clear tb prefs for that printer, reinstall and check what the units are?

I will try tomorrow evening.
1. uninstall driver
2. fresh tb installation
3. reinstall driver
4. look at tb prefs (**which one?**) just print_paper_size_unit ?
5. install PTNG
6. check units

that's okay or anything else?

**Comment by cleidigh** [comment]
@RRoenn 
That should be good.
I am going to try a change in how I save settings. We can try that tomorrow before you go through that, maybe my fix will work.
I will update you so you don't do anything unnecessary.
@cleidigh

**Comment by cleidigh** [comment]
@RRoenn 
@kiki-ja 
I think we hold off reinstalling drivers.
b18 now only saves margins, headers and footers. Theoretically I should never be able to change the units.
I would set the units to 1 for the printer that flipped, then keep changing the printer like before and see if it changes. Then check the console.
Also set print.Debug = true, that has some tb output.

b18 fixes table issue and has de tweaks.
@cleidigh

**Comment by RRoenn** [comment]
I did some quick checks on other PCs all with the same printer, some with really old TB/plugin.
As you expected, all values from print_paper_size_unit are set to 1.
Not all installed printer have a print_paper_size_unit value, e.g. 6 printer are installed, only 3 are shown.

1. PC
Windows 10 Pro (64bit), Version 21H2
TB 52.9.1 (32-Bit)
PT 1.3.0
print_paper_size_unit: all shown are 1

2.  PC
Windows 10 Pro (64bit), Version 21H2
TB 78.12.0 (64-Bit)
PTNG 2.0.61 / 2.1.1: v100+ Mai 2022
print_paper_size_unit: all shown are 1

3. PC
Windows 10 Pro (64bit), Version 21H2
TB 102.6.1
PTNG 2.1.5
print_paper_size_unit: all shown are 1

4.  PC
Windows 10 Pro (64bit), Version 22H2
TB 91.9.0 (32-Bit)
PTNG 2.1.5 / 2.1.1: v100+ Mai 2022
print_paper_size_unit: all shown are 1

**Comment by cleidigh** [comment]
@RRoenn 
Yes that is what I expected, however, I have to be honest the way that units are determined is very unclear. They are associated with the paper type, but seem to also be set according to both the driver and locale/os region.
If you use print.debug you can see the pageIds and associated paper units.

My hope is that the b18 change to only write back the margins,hdr ftr settings will prevent the units from getting charged.

I really appreciate the help since i know everyone has a day job!
@cleidigh

**Comment by RRoenn** [comment]
@cleidigh 
thank you for your continuous work. We use the addon every day at our day job ;-)

1. I set all print_paper_size_unit to 1
2. want to correct to wrong 322,58 to 12,7 - **can't input a , just a .**
 (after saving . was converted to a , )
4. while switching printer - table stays fine
5. looking for some little help at tab 5 - e.g. 
    &P = Page Number
    &D = Date Time
    &T = Message Title
    &U = Message Address (Local or URL)
    &PT = Page Number / Total number of pages
6. on tab 4 please change "Durchsuche" to "Durchsuchen" (missing n)
7. on tab 2 is it possible to get more contrast? see https://github.com/cleidigh/printing-tools-ng/issues/159#issuecomment-1419033186
8. looking at print preview - switching some printers - all looks okay
9. set mozilla_save_to_pdf back to 0 - switching some printers - all stays okay

thums up

greetings
@RRoenn

**Comment by RRoenn** [comment]
> Also set print.Debug = true, that has some tb output.

where do i have to set the value ?

**Comment by cleidigh** [comment]
@RRoenn 
It gives me great pleasure to work on something useful for people and I very much enjoy working with you to improve PTNG together.

Quick question then will follow with comments to all above.

Does it look like the units stay correct now?
@cleidigh

**Comment by RRoenn** [comment]
> Does it look like the units stay correct now?

all looks okay - all stays okay - so YES! :-)

**Comment by cleidigh** [comment]
@RRoenn 
I will comment using your numbers :  

2. Typical US stupidity on my part, the input field filter assumed only '.' decimal, I will add ','
3. Good, fixed
4. I have a table already in help, will be in b19. What I don't know is if I should or can put a help ? On tab 5, no good place. Any suggestions?
5. ok
6. This should never be undefined, can you set it?
7. I am trying to figure out what would give more contrast. It's black on blue with italics. I could not find any examples in the tb ui. Do you have any ideas?
8. These will be in help.

Ok so it looks like b18 fixed the table and units issue. Now I have to fix comma input and directory undefined.

Great feedback and testing.
@cleidigh

**Comment by cleidigh** [comment]
@RRoenn 
On # 6 can you check the preference:
extensions.printingtoolsng.pdf.enable_pdf_output_dir
when you see undefined 
So far I can't see how it goes to undefined 

@kiki-ja 
I am still working on the help and what are hopefully the last issues.
I did miss your css in b18, will have it in b19.

@tidebre67
We did have an issue with the paper_size_units
It was getting set to 0 sometimes. B18 fixes.

back to work... %-}
@cleidigh

**Comment by RRoenn** [comment]
at # 4 what do you think ?
**version 1**
![grafik](https://user-images.githubusercontent.com/77779094/217490675-e9b6ac14-5400-4e16-946a-c912e48b8112.png)


**version 2**
![grafik](https://user-images.githubusercontent.com/77779094/217491454-29947253-7c86-47d5-a12b-5ab9e006e019.png)


or mousehover ...

**Comment by RRoenn** [comment]
at # 7 what do you think ? (color is rgb 33,33,33 or #212121) or maybe a little more dark ...

![grafik](https://user-images.githubusercontent.com/77779094/217493918-dfc47168-a602-4164-a657-1289559c2a2e.png)

(old version)
![grafik](https://user-images.githubusercontent.com/77779094/217494075-96edd614-2241-463b-9946-8ce2b0e5510e.png)

**Comment by kiki-ja** [comment]
@cleidigh 

>I am still working on the help and what are hopefully the last issues.

Roger that.
I am working on translating the updated help.

>I did miss your css in b18, will have it in b19.

Roger.

@kiki-ja

**Comment by tidebre67** [comment]
@cleidigh , seems that the settings e.g. as below are not kept in print preview.
![Settings](https://user-images.githubusercontent.com/44371934/217588818-ffce39a3-99e0-4607-b7e9-a849bb302a41.png)
![margins](https://user-images.githubusercontent.com/44371934/217590682-82baf5a2-7b75-4915-a216-d93ac71b6155.png)

once a print is finished the marked margin jumps back e.g to 12.7 . 
I found that in destinations PDF, local printer apparently not affected. 
No idea whether this was already reported.

**Comment by cleidigh** [comment]
@tidebre67
When you say it jumps back do you mean the second time you do a preview?
@cleidigh

**Comment by tidebre67** [comment]
yes, and when i open a new file for a  print it is back to 12.7 instead 22.7 in the print preview.

**Comment by cleidigh** [comment]
@tidebre67
Is there a repeatable sequence I can try? So far my previews track ptng settings. 
@cleidigh

**Comment by cleidigh** [comment]
@tidebre67 
I guess part of what I am asking is what is the initial good setup and once there is an issue how do you get back to a correct state. 12.7mm = 0.5in is the default for margins so I presume its getting reset to defaults. But when in the sequence do you set it to 22.7?
@cleidigh

**Comment by cleidigh** [comment]
@RRoenn 
Did you change css in your contrast experiment?
I am trying to see where I can change it.
@cleidigh

**Comment by cleidigh** [comment]
@RRoenn 
How did you do your example, did you actually change the css?
Not sure how you arrived at your color, but #333333 is basically the background color and almost invisible. Here is a shot using #626262
Any darker and it gets hard to see.
What do you think?

<img width="411" alt="2023-02-08-14-14-40 " src="https://user-images.githubusercontent.com/25272315/217630968-932b1aa8-0cc6-4cd7-857f-931501e52d66.png">

@cleidigh

**Comment by tidebre67** [comment]
> @tidebre67 I guess part of what I am asking is what is the initial good setup and once there is an issue how do you get back to a correct state. 12.7mm = 0.5in is the default for margins so I presume its getting reset to defaults. But when in the sequence do you set it to 22.7? @cleidigh

@cleidigh :  
understand the 12.7mm and 0.5 inch are default.
changed the margin in the settings but in the print preview the default figures remain unchanged.
changed the margin in the print preview manually and printed the file with the updated margin, print ok.
opened the file again in the preview and the default figures are back.
really strange. checked the config but all looks correct with the updated margin.

**Comment by cleidigh** [comment]
@tidebre67 
Is it correct that the preview never shows changes from the settings?

**Comment by tidebre67** [comment]
yes, correct.

**Comment by cleidigh** [comment]
if you do a search in config on PDF24 you will see the margins 
then there will be one ptng entry containing the copies, ranges margins and hdrs ftrs
do the margins match?
what about hdrs and footers, if you set in ptng do they get printed correctly?

**Comment by RRoenn** [comment]
> @RRoenn How did you do your example, did you actually change the css? Not sure how you arrived at your color, but #333333 is basically the background color and almost invisible. Here is a shot using #626262 Any darker and it gets hard to see. What do you think?
> 
> <img alt="2023-02-08-14-14-40 " width="411" src="https://user-images.githubusercontent.com/25272315/217630968-932b1aa8-0cc6-4cd7-857f-931501e52d66.png">
> 
> @cleidigh

Thumbs up, I like it. I had used MS Paint ...

**Comment by cleidigh** [comment]
@RRoenn 
Great, will go with that.
Have you ever seen what @tidebre67 is seeing?
@cleidigh

**Comment by RRoenn** [comment]
> @RRoenn Great, will go with that. Have you ever seen what @tidebre67 is seeing? @cleidigh

just tried with my HP Colorjet, every is fine. But with PDF24 I get the same error, it is a pity.

and the next mysterium.
I set margins to 50.5 - go to print preview - go back to PTNG config - my 50.5 was changed to 1270.00 - and more and more

**Comment by tidebre67** [comment]
@cleidigh 
@RRoenn 
see more or less the same, but not only with PDF24.

**Comment by RRoenn** [comment]
next observation, I have set 2 copies - open print preview - for a short moment I see the 2 copies - but the setting is automatically switched back to 1 - every printer

**Comment by RRoenn** [comment]
> @cleidigh @RRoenn see more or less the same, but not only with PDF24.

what are your other printers? 

Greetings to Bremen :-)

**Comment by RRoenn** [comment]
@cleidigh 
I know we're fighting a lot of bugs right now, would it be foolish to ask if you could include scaling in the PTNG settings as well?

**Comment by tidebre67** [comment]
@cleidigh
@RRoenn
Header and footer are fine, the only issue i've found so far are the margins. Will revert with all printer namens tomorrow as i am out now and replying from mobil.
@RRoenn greetings too, guess somewhere in DE ?

**Comment by cleidigh** [comment]
@RRoenn 
I think I want to avoid adding, even 1 item is a whole lot of work and further complicates the tricky layout. The other thing is that we don't really gain since tb will retain that for every printer anyway.
@cleidigh

**Comment by cleidigh** [comment]
I guess we will need debug, I am confused...

**Comment by RRoenn** [comment]
@cleidigh 
take a nap. I know you will find the clue

maybe somewhere where you made an extra rule for printers with "pdf" in their name. A tab or bracket will be missing...

**Comment by RRoenn** [comment]
maybe we can track the margins issue here #162 ?

this issue is quite long and has many topics ...

**Comment by kiki-ja** [comment]
@cleidigh
@RRoenn
@tidebre67
and All.

Traffic control and requests.

This is a **'New UI'** issue.
If you have an issue related to **'Page options'**, please comment or report it in #139.
If you have an issue related to **'PDF Output (PDF File Naming)'**, please comment or report it in #151.

The reason for this is that it is confusing to the reader.

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
@tidebre67 
@RRoenn 
I barely got b19 posted : 
b19 - debug, de tweak, ja css, undefined dir fix, list contrast, copies fix 1

we should look at the console entering and leaving ui
I print margins in prefs and our ptng settings 
we should see when they change
See how the copies look.
@cleidigh

**Comment by cleidigh** [comment]
Help for tab 5
I  put help icon at top right 
this will point to tab help
Not perfect, but best option 
@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

PTNG 2.2.0-b19
Thunderbird 102.7.2 64bit
Windows 10 Pro 22H2 64bit

>Help for tab 5
>I put help icon at top right

I find the right side of 'Page Options' appropriate.
Please see screenshot.

![220b19-tab5-helpicon](https://user-images.githubusercontent.com/80428201/217705166-709dddc8-1346-4cb9-bfbc-0e0c62ddb927.png)

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh

PTNG 2.2.0-b19
Thunderbird 102.7.2 64bit
Windows 10 Pro 22H2 64bit

I moved the help icon.
How do you like it?

![220b19-tab5-help-icon](https://user-images.githubusercontent.com/80428201/217978240-e4389fad-a147-48cf-a87f-7d809527c5ea.png)

[Attach xhtml file]
[tab5-ptng-options.xhtml.zip](https://github.com/cleidigh/printing-tools-ng/files/10702930/tab5-ptng-options.xhtml.zip)

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
That would be ok if we were only giving help for page options.
I want the icon top and right because it's help for the whole tab
Maybe we do something else for all help some other time.
Not perfect,
@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

Understood.
I will wait for the help to be completed.

@kiki-ja
